storage/innobase/include/ut0ut.h

```diff
@@ -88,15 +88,26 @@ struct ut_when_dtor {
    the YieldProcessor macro defined in WinNT.h. It is a CPU architecture-
    independent way by using YieldProcessor. */
 #  define UT_RELAX_CPU() YieldProcessor()
-# elif defined(HAVE_ATOMIC_BUILTINS)
+# elif defined(__powerpc__)
+#include <sys/platform/ppc.h>
 #  define UT_RELAX_CPU() do { \
-     volatile lint	volatile_var; \
-     os_compare_and_swap_lint(&volatile_var, 0, 1); \
```

Comment:
Because of UT_COMPILER_BARRIER the loop won't be optimised away and there will just be an empty loop.

I've no objection to leaving the HAVE_ATOMIC_BUILTINS implementation for non x86/powerpc architectures until they can come up with something better. I was just following the MySQL upstream patch in this respect.